### PR TITLE
Blob rounds end when blob dies

### PR DESCRIPTION
**Pull request by Robustin**
Seriously who thought it was a good idea to keep the round going after a blob is defeated?

### CODE REVIEW

**Comment by PKPenguin321** [comment]
thank you for this
blob rounds should only continue if the blob dies in like the first 3 minutes after it pops or something

File: code/game/gamemodes/blob/blob_finish.dm [hunk 1]
```diff
@@ -4,16 +4,14 @@
...
-			return ..()
-		return 1
-	return ..()
+		continuous_sanity_checked = 1 //Nonstandard definition of "alive" gets past the check otherwise
+		SSshuttle.emergencyNoEscape = 0
+		if(SSshuttle.emergency.mode == SHUTTLE_STRANDED)
+			SSshuttle.emergency.mode = SHUTTLE_DOCKED
+			SSshuttle.emergency.timer = world.time
+			priority_announce("Hostile enviroment resolved. You have 3 minutes to board the Emergency Shuttle.", null, 'sound/AI/shuttledock.ogg', "Priority")
+	return 1
```
Comment (Shadowlight213) [inline code/game/gamemodes/blob/blob_finish.dm]: This is going to end the round EVERY time this check is ran.
Regardless of if the blobs are dead or anything. the round will immediately end.

**Comment by Shadowlight213** [comment]
What is this even supposed to do?
Are you trying to remove the config option? Just message me and I can disable it instead of trying to force shit through with broken code changes.

**Comment by Robustin** [comment]
I mean that's precisely what it does. Before it would force a return if continuous was on, which it always is. Now it goes through the steps for the shuttle in case its not blob mode (at least I'm pretty sure thats why that exists) and returns 1 regardless.

**Comment by Incoming5643** [comment]
If you return 1 the round ends. If you return ..() it checks the generic conditionals for round ends and runs mulligans if it feels like it's a good round for it. Returning 0 would keep a round going forever.

In a perfect world it should always return ..() but since camera mobs follow their own rules the check is here.

#17526 is the proper fix to this problem, you're just removing functionality with this.

**Comment by Incoming5643** [comment]
Also it's a good idea because people in blob rounds all work under the assumption that "nothing matters" besides killing the blob. 

When the blob dies and the command staff suddenly has to deal with the reality that they gave a bunch of people (some of whom are now probably antags) guns and all access I consider that !!FUN!! and an important lesson in not treating blob rounds in a vacuum.

**Comment by Robustin** [comment]
Sure if someone has a better solution go for it, I just made this PR because it seemed like an easy fix.

File: code/game/gamemodes/blob/blob_finish.dm (hunk 1 above)
Comment (Robustin) [inline code/game/gamemodes/blob/blob_finish.dm]: wew-ps

File: code/game/gamemodes/blob/blob_finish.dm [hunk 2]
```diff
@@ -4,16 +4,14 @@
...
-				SSshuttle.emergency.timer = world.time
-				priority_announce("Hostile enviroment resolved. You have 3 minutes to board the Emergency Shuttle.", null, 'sound/AI/shuttledock.ogg', "Priority")
-			return ..()
+		continuous_sanity_checked = 1 //Nonstandard definition of "alive" gets past the check otherwise
+		SSshuttle.emergencyNoEscape = 0
+		if(SSshuttle.emergency.mode == SHUTTLE_STRANDED)
+			SSshuttle.emergency.mode = SHUTTLE_DOCKED
+			SSshuttle.emergency.timer = world.time
+			priority_announce("Hostile enviroment resolved. You have 3 minutes to board the Emergency Shuttle.", null, 'sound/AI/shuttledock.ogg', "Priority")
 		return 1
```
Comment (Shadowlight213) [inline code/game/gamemodes/blob/blob_finish.dm]: This is going to say hostile environment resolved, then immediately end the round.
Returning 1 means end the round immediately.

**Comment by Cheridan** [comment]
i'll close this then as per incoming's above comments

**Pull request closed without merging**